include/quda.h

```diff
@@ -185,6 +185,10 @@ extern "C" {
 
     int num_src; /**< Number of sources in the multiple source solver */
 
+    /**< The grid of sub-partition according to which the processor grid will be partitioned
+    The product of the four integers should be equal to `num_src`. */
+    int split_grid[QUDA_MAX_DIM];
+
```

Comment:
We should be future proofing here, and think about where we have a block or psuedo-block solver per sub-partition.  So `num_src = num_src_block * num_src_grid` or something like that, where `num_src_grid` is the product of `split_grid` and represents the number of (block)solves done in parallel over sub-partitions, and `num_src_block` is the number of solves per sub-partition.